database/database.go

```diff
@@ -11,6 +11,12 @@ package database
 type Database interface {
 	DataAccessor
 
+	// DeleteUpToLocation deletes as much data as it can from the given store, while guaranteeing
+	// that the data belongs to dbLocation, its following locations and dbPreservedLocations will
+	// be kept.
```

Comment:
Please rephrase it to "while keeping the data that dbLocation, its following locations, and dbPreservedLocations point to."

Reply:
Also, make sure to propagate this change to everywhere else this comment is written.